### DISCUSSION TITLE
Constraints in the initializer for object_permissions.yml

### DISCUSSION THREAD

**Discussion by mepramo** [netbox-community/netbox-docker#636] [Ideas]
Hi Team,

I wanted to add some constraints into the object_permissions.yml which is part of the initializer scripts.

The idea was to do minimum changes from the GUI and create multiple templated permissions for each group/tenant inside the yaml.

I could not make it work. Looks like the file 015_object_permission.py does not have a handling given for the constraints.

Is this a feature coming soon ?

**Comment by Lon1** [comment]
Needed to add a bit of code to 015_object_permissions.py to the end and add import of json at the top:
```
import json
.
.
.
    if permission_details.get("constraints", 0):
        object_permission.constraints = None
        constraint_line = ""
        for constraint_detail in permission_details["constraints"]:
            constraint_line = constraint_line + constraint_detail

        if constraint_line:
            object_permission.constraints = json.loads(constraint_line)
            print(
                    "🔓 Assigned constraint of %s "
                    % (object_permission.constraints)
                )
```
Then in object_permissions.yml can contain any of these variations (as examples):
```
constraints:
      - '{"custom_field_data__support_group__in": ['
      - '"Network-Mgmt",'
      - '"Network-Security"]}'

constraints:
        - '[{"custom_field_data__support_group": "Network-Core"},{"custom_field_data__support_group": "Network-Security"}]'

constraints:
        - '{"custom_field_data__support_group": "Network-Core"}'
```
